configure/os/CONFIG.win32-x86.win32-x86

```diff
@@ -185,7 +185,9 @@ RES=.res
 
 # MS Visual C++ doesn't recognize *.cc as a C++ source file,
 # so C++ compiles get the flag -TP
-COMPILER_CXXFLAGS = -TP
+# We want MSVC to report a reasonable C++ version in the __cplusplus macro so
+# we need to set -Zc:__cplusplus (this is at least required for MSVC 19.xx).
+COMPILER_CXXFLAGS = -TP -Zc:__cplusplus
```

Comment:
I hadn't come across the [`-Zc:*` family of options](https://docs.microsoft.com/en-us/cpp/build/reference/zc-conformance?view=msvc-160) before.  I had come across [`/std=*`](https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-160), but that is >=2017.

[-Zc:__cplusplus](https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-160) is documented as >=2017 (15.7).  Since the older appveyor builds succeed without additional warnings, I guess the older compilers are silently ignoring unknown `-Zc:*` operations?

And hey, with [`/Zc:preprocessor`](https://docs.microsoft.com/en-us/cpp/build/reference/zc-preprocessor?view=msvc-160) the MS preprocessor is finally come up to [C99 conformance](https://devblogs.microsoft.com/cppblog/msvc-preprocessor-progress-towards-conformance/).